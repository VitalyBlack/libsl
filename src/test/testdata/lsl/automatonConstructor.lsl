libsl "1.0.0";
library simple;

types {
   Int(int);
   String(string);
}

automaton A(
   var i: Int,
   var s: String
) {
   var i: Int;

   fun func() {
      i = new B(state = s1, v = 0);
   }
}

automaton B(
   var v: Int
) {
   state s1;
}