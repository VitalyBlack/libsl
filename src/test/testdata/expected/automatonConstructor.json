{
  "name": "simple",
  "lslVersion": "1.0.0",
  "types": [
    {
      "name": "Int",
      "realName": "int"
    },
    {
      "name": "String",
      "realName": "string"
    }
  ],
  "automata": [
    {
      "name": "A",
      "constructorVariables": [
        {
          "name": "i",
          "type": "Int"
        },
        {
          "name": "s",
          "type": "String"
        }
      ],
      "variables": [
        {
          "name": "i",
          "type": "Int"
        }
      ],
      "states": [],
      "shifts": [],
      "functions": [
        {
          "name": "func",
          "automaton": "A",
          "args": [],
          "contracts": [],
          "statements": [
            {
              "kind": "assignment",
              "variable": "A.i",
              "value": {
                "kind": "callAutomatonConstructor",
                "automatonName": "B",
                "state": "s1",
                "args": [
                  {
                    "name": "v",
                    "value": {
                      "kind": "integer",
                      "value": 0
                    }
                  }
                ]
              }
            }
          ]
        }
      ]
    },
    {
      "name": "B",
      "constructorVariables": [
        {
          "name": "v",
          "type": "Int"
        }
      ],
      "variables": [],
      "states": [
        {
          "name": "s1",
          "kind": "SIMPLE"
        }
      ],
      "shifts": [],
      "functions": []
    }
  ]
}